src/core/pointcloud/qgspointcloudattributemodel.cpp

```diff
@@ -0,0 +1,237 @@
+/***************************************************************************
+                         qgspointcloudattributemodel.cpp
+                         ---------------------
+    begin                : November 2020
+    copyright            : (C) 2020 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgspointcloudattributemodel.h"
+#include "qgspointcloudlayer.h"
+#include "qgspointcloudindex.h"
+#include "qgsapplication.h"
+
+QgsPointCloudAttributeModel::QgsPointCloudAttributeModel( QObject *parent )
+  : QAbstractItemModel( parent )
+{
+
+}
+
+void QgsPointCloudAttributeModel::setLayer( QgsPointCloudLayer *layer )
+{
+  if ( layer && layer->dataProvider() && layer->dataProvider()->index() )
+  {
+    mLayer = layer;
+    setAttributes( layer->dataProvider()->index()->attributes() );
```

Comment:
thinking about attributes() in long term (e.g. with PDAL provider). the index may not be available (or under construction), but attributes would  be from the PDAL header. wondering if we shouldn't add attributes() call to dataProvider() or even QgsPointCloudLayer .?

Reply:
I agree, and would suggest we add it to both. But I'll do that in a follow up pr.